mautic/mautic
# Salesforce contacts dnc sync fix

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | Y
| New feature/enhancement? (use the a.x branch)      | N
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | Y 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
We need to synchronize DNC record for new Lead/Contacts downloaded from SalesForce. Currently the plugin relied solely on DNC history, which is not synchronized unless we push something and nor it is synchronized if contact has been created opted-out.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Setting up SF plugin
3. Selecting “Use latest updated Do Not Contact record” under “Features” in Salesforce plugin
4. Pulling contacts from SF who are labeled as DNC out of the box into Mautic.
5. Verify downloaded is DNCed

One more testing is required in this PR :-

1. Create a new Lead in Salesforce and run the command php app/console mautic:integration:synccontacts -i Salesforce and note the contact exists in Mautic
2. Delete the contact in Mautic. Update a mapped field for the Lead in SF (i.e. last name) then delete them in SF.
3. Run the command php app/console mautic:integration:synccontacts -i Salesforce again
4. The contact should not be recreated in Mautic
